pkg/resolver/search.go

```diff
@@ -70,18 +70,28 @@ func (s *SearchResult) Items() []map[string]interface{} {
 
 func (s *SearchResult) Related() []SearchRelatedResult {
 	klog.V(2).Info("Resolving SearchResult:Related()")
-
-	// FIXME: WORKAROUND when the query doesn't request Items() we must use a more efficient query to get the uids.
 	if s.uids == nil {
-		klog.Warning("TODO: Use a query that only selects the UIDs.")
-		s.Items()
+		s.Uids()
 	}
-	s.wg.Wait() // WORKAROUND wait to complete execution of Items()
+
+	s.wg.Wait()
 
 	r := s.getRelations()
 	return r
 }
 
+func (s *SearchResult) Uids() []*string {
```

Comment:
No need to return the `[]*string` because we are saving the result in the SearchResult object.